src/signals/incident/components/IncidentForm/index.test.js

```diff
@@ -195,9 +201,7 @@ describe('<IncidentForm />', () => {
           next,
         });
 
-        defer(() => {
-          expect(next).toHaveBeenCalled();
-        });
+        expect(next).toHaveBeenCalled();
```

Comment:
You should also assert that `next` has not been called prior to setting `instance.form.valid` to 'true'.